app/controllers/patients_controller.rb

```diff
@@ -249,53 +249,170 @@ def update
     render json: patient
   end
 
-  def update_hoh
+  # Moves a record into a household
+  def move_to_household
     new_hoh_id = params.permit(:new_hoh_id)[:new_hoh_id]&.to_i
-    current_patient_id = params.permit(:id)[:id]
-    household_ids = params[:household_ids] || []
+    current_patient_id = params.permit(:id)[:id]&.to_i
+
     current_patient = current_user.get_patient(current_patient_id)
-    old_hoh = current_user.get_patient(current_patient.responder_id)
     new_hoh = current_user.get_patient(new_hoh_id)
+    current_user_patients = current_user.patients
+
+    # ----- Error Checking -----
+
+    # Check to make sure selected HoH record exists.
+    unless Patient.where(purged: false).exists?(new_hoh_id)
+      error_message = "Move to household action failed: Selected Head of Household with ID #{new_hoh_id} does not exist."
+      render(json: { error: error_message }, status: :forbidden) && return
+    end
 
-    # Don't do anything if there hasn't been a change to the responder at all
+    # Check to make sure user has access to update this record.
+    unless current_user_patients.exists?(current_patient_id)
+      error_message = 'Move to household action failed: User does not have permissions to update current monitoree.'
+      render(json: { error: error_message }, status: :forbidden) && return
+    end
+
+    # Do not do anything if there hasn't been a change to the responder at all.
     redirect_to(root_url) && return if current_patient.responder_id == new_hoh_id
```

Comment:
I know we typically redirect to `root_url` for a lot of cases, but would it make sense to just redirect to the patient page that the user was on before instead of redirecting to the home (linelist) page?

Reply:
That is actually what it does as it is right now. You can test it out by clicking "Move to Household" and selecting the current record. Though I am a little unclear as to how that works given the `root_url` is technically not the patient page.  Looking into why exactly. Let me know if you have any thoughts.